public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx

```diff
@@ -6,372 +6,321 @@
 
 export interface AssociatedObjectsTabProps {
-  associatedObjects: AssociatedObject[];
+  datasource: DatasourceDetails;
 }
 
-interface FilterOption {
-  value: string;
-  text: string;
-}
+export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = (props) => {
+  const { datasource } = props;
+  const [selectedDatabase, setSelectedDatabase] = useState<string>('');
+  const [isRefreshing, setIsRefreshing] = useState<boolean>(false);
+  const [lastUpdated, setLastUpdated] = useState(new Date().toLocaleString());
+  const [isObjectsLoading, setIsObjectsLoading] = useState<boolean>(false);
+  const [cachedDatabases, setCachedDatabases] = useState<CachedDatabase[]>([]);
+  const [cachedTables, setCachedTables] = useState<CachedTable[]>([]);
+  const [cachedAccelerations, setCachedAccelerations] = useState<CachedAcceleration[]>([]);
+  const [associatedObjects, setAssociatedObjects] = useState<AssociatedObject[]>([]);
+  const [isFirstTimeLoading, setIsFirstTimeLoading] = useState<boolean>(true);
 
-interface AssociatedTableFilter {
-  type: string;
-  field: string;
-  operator: string;
-  value: string;
-}
+  const {
+    loadStatus: databasesLoadStatus,
+    startLoading: startLoadingDatabases,
+  } = useLoadDatabasesToCache();
+  const { loadStatus: tablesLoadStatus, startLoading: startLoadingTables } = useLoadTablesToCache();
+  const {
+    loadStatus: accelerationsLoadStatus,
+    startLoading: startLoadingAccelerations,
+  } = useLoadAccelerationsToCache();
 
-export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = ({
-  associatedObjects,
-}) => {
-  const [lastUpdated, setLastUpdated] = useState('');
-  const [databaseFilterOptions, setDatabaseFilterOptions] = useState<FilterOption[]>([]);
-  const [accelerationFilterOptions, setAccelerationFilterOptions] = useState<FilterOption[]>([]);
-  const [filteredObjects, setFilteredObjects] = useState<AssociatedObject[]>([]);
+  let lastChecked: boolean;
+  if (selectedDatabase !== '') {
+    lastChecked = true;
+  } else {
+    lastChecked = false;
+  }
+  // Get last selected if there is one, set to first option if not
+  const [databaseSelectorOptions, setDatabaseSelectorOptions] = useState(
+    cachedDatabases.map((database, index) => {
+      return {
+        label: database.name,
+        checked: lastChecked
+          ? database.name === selectedDatabase
+            ? 'on'
+            : index === 0
+            ? 'on'
+            : undefined
+          : undefined,
+      };
+    })
+  );
 
-  // TODO: FINISH THE REFRESH LOGIC
-  const fetchAssociatedObjects = async () => {
-    // Placeholder for data fetching logic
-    // After fetching data:
-    // setAssociatedObjects(fetchedData);
-    const now = new Date();
-    setLastUpdated(now.toUTCString()); // Update last updated time
+  const onRefreshButtonClick = () => {
+    startLoadingDatabases(datasource.name);
+    startLoadingTables(datasource.name, selectedDatabase);
```

Comment:
Loading tables will error out if the database is not valid anymore? 
Also, will we run in race conditions? If the result of databases comes after the result of tables. I would Ideally want to load databases first and then load tables and acceleration in parallel.

Reply:
Will create a task for fast-followup to this.